src/Microsoft.Identity.Web/CertificateManagement/DefaultCertificateLoader.cs

```diff
@@ -174,11 +174,7 @@ private static void ParseStoreLocationAndName(string storeDescription, ref Store
         /// <summary>
         /// Find a certificate by criteria.
         /// </summary>
-        /// <param name="x509Store"></param>
-        /// <param name="identifierCriterium"></param>
-        /// <param name="certificateIdentifier"></param>
-        /// <returns></returns>
-        private static X509Certificate2 FindCertificateByCriterium(
+        private static X509Certificate2? FindCertificateByCriterium(
```

Comment:
Wondering if instead of returning `null` we throw an exception in this method, if the certificate is not found. Other load methods will throw exceptions if the certificate can't be created.

Reply:
the keyvault one will throw an exception, and if parsing the location fails, we throw an exception, otherwise we are returning the cert or null on all the other load methods.

---
In reply to: [444030354](https://github.com/AzureAD/microsoft-identity-web/pull/231#discussion_r444030354) [](ancestors = 444030354)

Reply:
If the certificate is not found, FromDisk, FromBase64, and FromKeyVault throw an exception; two FromCertStore methods return null. So my thought was to make the behavior consistent. Perhaps log that cert was not found and return null? This might benefit the certificate rotation logic (when implemented) when multiple certs are specified and we want to try until valid one is found.